git-cliff-core/src/repo.rs

```diff
@@ -95,11 +96,47 @@ impl Repository {
 	/// Returns the current tag.
 	///
 	/// It is the same as running `git describe --tags`
-	pub fn current_tag(&self) -> Option<String> {
+	pub fn current_tag(&self) -> Option<Tag> {
 		self.inner
 			.describe(DescribeOptions::new().describe_tags())
 			.ok()
-			.and_then(|describe| describe.format(None).ok())
+			.and_then(|describe| {
+				describe
+					.format(None)
+					.ok()
+					.map(|name| self.resolve_tag(&name))
+			})
+	}
+
+	/// Returns the tag object of the given name.
+	/// if given name don't exists, still returns Tag object with the given name
+	pub fn resolve_tag(&self, name: &str) -> Tag {
+		match self.inner.resolve_reference_from_short_name(name) {
+			Ok(reference) => match reference.peel_to_tag() {
+				Ok(tag) => Tag {
+					name:    tag.name().unwrap_or_default().to_owned(),
+					message: tag.message().map(Self::cleanup_message),
+				},
+				_ => Tag {
+					name:    name.to_owned(),
+					message: None,
+				},
+			},
+			_ => Tag {
+				name:    name.to_owned(),
+				message: None,
+			},
+		}
+	}
+
+	/// used to remove signature from signed tag message
+	fn cleanup_message(message: &str) -> String {
+		let re = Regex::new(
+			r"(?s)-----BEGIN PGP SIGNATURE-----(.*?)-----END PGP SIGNATURE-----",
+		)
+		.expect("invalid regex, wtf");
```

Comment:
Let's use a lazy regex here.

e.g.

https://github.com/orhun/git-cliff/blob/7f594e04255a0ba697071e8e1b49e0f74d6ef4d9/git-cliff-core/src/commit.rs#L36

Then we can also remove `cleanup_message` function and simply use the regex in `resolve_tag`.

Reply:
cool! done